cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java

```diff
@@ -169,6 +171,7 @@ private static String getSwitchNode2Id(VoltageLevel vl, Switch sw) {
 
     private static void writeSubstations(Network network, String cimNamespace, XMLStreamWriter writer) throws XMLStreamException {
         Map<String, String> geographicalRegionIds = new HashMap<>();
+        BiMap<String, String> subGeographicalRegionIds = HashBiMap.create();
         for (Substation substation : network.getSubstations()) {
             String geoName = network.getNameOrId();
```

Comment:
Proposal to reduce the complexity of code, separate pieces into two small functions that could be called like:

```java
geographicalRegionIds.computeIfAbsent(geographicalRegionName(substation),
    geoName -> writeGeographicalRegion(geoName, cimNamespace, writer));
```
With this I would like to:
- encapsulate how to obtain country name from substation or network, 
- avoid declaring a `finalGeoName`.
- and move the try/catch to a separate code block.